### PR TITLE
Rst 6473 ut validate phone number

### PR DESCRIPTION
**Pull request by itoulou**
https://tools.hmcts.net/jira/browse/RST-6473

### CODE REVIEW

File: tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            valid_numbers = ['0', '123', '34324', '07111111111', '07123456789', '31341432131432143124']
+            for number in valid_numbers:
+                form.phone.data = number
+                assert validate_phone_number(form, form.phone) is None
+
+    def test_validate_phone_number_invalid_non_numeric(self, app):
+        with app.app_context():
+            form = ContactPreferencesForm()
+            form.contact_options.data = 'PHONE'
+            valid_numbers = ['0A', '12dad3', '34DSAD324', '07111ddsaDQ111dW111d', '071234SADDQD', '432DSAD143124']
```
Comment (mark-butler-solirius) [inline tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py:22]: name of array confusing

File: tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            for number in valid_numbers:
+                form.phone.data = number
+                with pytest.raises(ValidationError, match='Enter a valid phone number'):
+                    validate_phone_number(form, form.phone)
+
+    def test_validate_phone_number_invalid_special_char(self, app):
+        with app.app_context():
+            form = ContactPreferencesForm()
+            form.contact_options.data = 'PHONE'
+            valid_numbers = ['0£', '1$23', '34*(324', '07.111111111', '071234+56789', '+3134143213', '+447788991122']
```
Comment (mark-butler-solirius) [inline tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py:32]: same

File: tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py (hunk 1 above)
Comment (itoulou) [reply tests/grc/unit/utils/test_custom_form_validators/test_validate_phone_number.py]: renamed to invalid_numbers

**Review by mark-butler-solirius — APPROVED** [review]
lg

**Pull request merged**